src/components/GameView/GameOverDialog.vue

```diff
@@ -46,16 +70,20 @@ export default {
         // do nothing - parent controls whether dialog is open
       },
     },
+    ...mapState({
+      playingFromDeck: ({ game }) => game.playingFromDeck,
+      game: ({ game }) => game,
+    }),
+    ...mapGetters(['player', 'opponent']),
     heading() {
-      if (this.stalemate) {
-        return 'Stalemate';
+      if (this.currentMatchWinner) {
+        return this.playerWinsMatch ? 'You Win the Match' : 'You Lose the Match';
       }
```

Comment:
It seems like this is a little unclear to me. We're checking if there's a winner at all and then switching based on that first? Otherwise we then check... sort of the same stuff below? Maybe I'm missing something but I'd prefer this to be much more straightforward with the logic and perhaps use a map of conditions, or a switch instead (I hate you for using them @itsalaidbacklife, lol, but it makes sense here).

Reply:
The tricky bit here is that we want to do different stuff based on whether the match is over and also whether you won or not. That means there's a few different variables which makes it difficult to use a switch since we have different things to compare.

I think this might be easier to understand if we rename `currentMatchWinner` to `matchIsOver` and `playerWins` to `playerWinsGame`. Idea being to differentiate the stuff about the match from the stuff about the latest game, both of which factor into the dialog header